ts/src/program/namespace/account.ts

```diff
@@ -187,12 +188,24 @@ export class AccountClient<T = any> {
 
   /**
    * Returns all instances of this account type for the program.
+   *
+   * @param filters User-provided filters to narrow the results from `connection.getProgramAccounts`.
+   *
+   *                When filters are not defined this method returns all
+   *                the account instances.
+   *
+   *                When filters are of type `Buffer`, the filters are appended
+   *                after the discriminator.
+   *
+   *                When filters are of type `GetProgramAccountsFilter[]`,
+   *                filters are appended after the discriminator filter.
    */
-  async all(filter?: Buffer): Promise<ProgramAccount<T>[]> {
-    let bytes = AccountsCoder.accountDiscriminator(this._idlAccount.name);
-    if (filter !== undefined) {
-      bytes = Buffer.concat([bytes, filter]);
-    }
+  async all(
```

Comment:
Let's add an integration test for this.

Reply:
I may need a bit of guidance for that, but would gladly do it

Reply:
Should it be like [this one](https://github.com/project-serum/anchor/blob/master/ts/tests/events.spec.ts)? I tested in my own app but I'll rather have a test for that.

Reply:
It would be easiest to add it to one of the `/tests`, e.g., `/tests/misc` as an end to end test.

Reply:
Gotcha, will work on it.